src/iommu/vfio.c

```diff
@@ -213,6 +210,26 @@ static int vfio_configure_iommu(struct vfio_container *vfio)
 	return 0;
 }
 
+static int vfio_group_set_container(struct vfio_group *group, struct vfio_container *vfio)
+{
+	log_info("adding group '%s' to container\n", group->path);
+
+	if (ioctl(group->fd, VFIO_GROUP_SET_CONTAINER, &vfio->fd)) {
+		log_debug("failed to add group to vfio container\n");
+		return -1;
+	}
+
+	if (vfio_configure_iommu(vfio)) {
+		log_error("failed to configure iommu\n");
+
+		log_fatal_if(ioctl(group->fd, VFIO_GROUP_UNSET_CONTAINER), "unset container\n");
```

Comment:
nice!. That answers my initial question that I left on the code.

Reply:
Yeah, and we want to consider it fatal if we cant unset here. If we couldn't unset, I dont know what the application or library should do to fix it. Something worse must be wrong.